### DISCUSSION TITLE
Simulating Worker/Queue Errors and Stalled Jobs

### DISCUSSION THREAD

**Discussion by lockenj** [taskforcesh/bullmq#2226] [Q&A]
First off BullMQ is AWESOME!!! Thanks for everyones hard work making this repo great!

I am introducing the Worker paradigm (via BullMQ) into our ecosystem. We have a handful of scenarios that need to guarantee that failed Jobs are restarted as fast as possible. 

To  ensure that our needs our met we are building an integration test suite that simulates worst case scenarios. 

I am looking for guidance on how to write an integration test that simulates:

1.  "Stalled" workers
2. A Worker throwing Error. In other words setup a scenario that will cause the Worker to emit [Error](https://api.docs.bullmq.io/interfaces/v4.WorkerListener.html#error)
3. A Queue throwing Error. In other words setup a scenario that will cause the Queue to emit [Error](https://api.docs.bullmq.io/interfaces/v4.QueueEventsListener.html#error)

Any guidance is much appreciated!

**Comment by manast** [comment]
You should be able to find such scenarios in the tests: https://github.com/taskforcesh/bullmq/tree/master/tests
